### ISSUE TITLE
Math not displaying correctly in accordion

### ISSUE THREAD

**Issue by rgouveiamendes** [marimo-team/marimo#2216]
### Describe the bug

Inline math is not displaying correctly in the summary of an accordion like in here:
![image](https://github.com/user-attachments/assets/b67066c4-8216-462d-8412-4fb785074cbb)

### Environment

```json
{
  "marimo": "0.8.9",
  "OS": "Linux",
  "OS Version": "6.9.3-76060903-generic",
  "Processor": "x86_64",
  "Python Version": "3.10.12",
  "Binaries": {
    "Browser": "--",
    "Node": "--"
  },
  "Requirements": {
    "click": "8.1.7",
    "importlib-resources": "missing",
    "jedi": "0.19.1",
    "markdown": "3.7",
    "pymdown-extensions": "10.9",
    "pygments": "2.18.0",
    "tomlkit": "0.13.2",
    "uvicorn": "0.30.6",
    "starlette": "0.38.4",
    "websockets": "13.0.1",
    "typing-extensions": "4.12.2",
    "ruff": "0.6.3"
  }
}
```

### Code to reproduce

```python
mo.accordion(
    {
        r"""**a)** Estime os coeficientes da regressão de $Y$ sobre $X_1$, bem como o erro padrão da regressão e o $R^2$. O que pensa da estimativa de $\beta_1$?""": mo.md(
            """
            - Introduzindo o comando `regress Y X1` obtemos a estimação abaixo
            - A estimativa para $\beta_1$ é inesperada, pois contradiz a teoria económica de que existe uma relação negativa entre o preço e as vendas.
            """
        )
    }
)
```

**Comment by rgouveiamendes** [comment]
This is not entirely solved yet.
When the output expands to more than a single line, the problem persists as in here:
![image](https://github.com/user-attachments/assets/d9117862-2678-4308-a938-694ce4741c1d)

MWE:
```python
mo.accordion(
    {
        """**b)** Se o valor esperado de $X$ for a média dos seus dois valores referidos na alínea anterior, qual acha que será o valor esperado de $Y$? Confirme a sua resposta usando a lei das expectativas iteradas.""": mo.md("""        
		$$\\mathbb{{E}}\\left[Y \\middle| X = \\frac{{800 + 1400}}{{2}}\\right] = \\mathbb{{E}}\\left[Y | X = 1100\\right] = 0.7 + 0.002 \\times 1100$$

        Podemos confirmar a lei das expectativas iteradas:

		$$\\begin{{aligned}}
    	\\mathbb{{E}}(Y) &= \\mathbb{{E}}\\left[\\mathbb{{E}}\\left[Y | X\\right]\\right] =\\\\
    	&= \\mathbb{{E}}\\left[0.7 + 0.002 X\\right] = \\\\
    	&= 0.7 + 0.002 \\, \\mathbb{{E}}\\left[X\\right]
    	\\end{{aligned}}$$
        """)
    }
)
```

**Comment by mscolnick** [comment]
Should be fully fixed by #2271 - thanks for the repro snippets
